validator/client/beacon-api/json_rest_handler.go

```diff
@@ -89,9 +89,12 @@ func decodeJsonResp(resp *http.Response, responseJson interface{}) (*apimiddlewa
 	if resp.StatusCode != http.StatusOK {
 		errorJson := &apimiddleware.DefaultErrorJson{}
 		if err := decoder.Decode(errorJson); err != nil {
-			return nil, errors.Wrapf(err, "failed to decode error json for %s", resp.Request.URL)
+			if resp.StatusCode == http.StatusNotFound {
+				errorJson = &apimiddleware.DefaultErrorJson{Code: http.StatusNotFound, Message: "Resource not found"}
+			} else {
+				return nil, errors.Wrapf(err, "failed to decode error json for %s", resp.Request.URL)
```

Comment:
can we do a dump here what was trying to decode?

Reply:
I am not 100% sure about this, but I think that after `decoder.Decode` reads the input, it's not longer available for reading.